src/model_ode.cpp

```diff
@@ -42,9 +42,17 @@ namespace amici {
         auto x_pos = computeX_pos(x);
         fdwdx(t, N_VGetArrayPointer(x_pos));
         SUNMatZero(J);
-        fJSparse(static_cast<SUNMatrixContent_Sparse>(SM_CONTENT_S(J)), t,
-                 N_VGetArrayPointer(x_pos), unscaledParameters.data(),
-                 fixedParameters.data(), h.data(), w.data(), dwdx.data());
+        try {
+            fJSparse(SM_DATA_S(J), t, N_VGetArrayPointer(x_pos),
+                     unscaledParameters.data(), fixedParameters.data(),
+                     h.data(), w.data(), dwdx.data());
+            fJSparse_colptrs(SM_INDEXPTRS_S(J));
+            fJSparse_rowvals(SM_INDEXVALS_S(J));
+        } catch (std::invalid_argument &) {
```

Comment:
Again somewhat abusive. Can't we check some type or add some model flag instead? Or is this planned to change sometime soon?